From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SoSeDiK <mrsosedik@gmail.com>
Date: Fri, 3 Nov 2023 22:48:21 +0200
Subject: [PATCH] Call BlockFormEvent for composter

Might not be the best event for it, but meh, it works

diff --git a/src/main/java/net/minecraft/world/level/block/ComposterBlock.java b/src/main/java/net/minecraft/world/level/block/ComposterBlock.java
index 27ff4a3de2253177a6f4a0dbf131d437eb24f2e8..62a222353695d6fbd44a97b9d0195a1106625815 100644
--- a/src/main/java/net/minecraft/world/level/block/ComposterBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/ComposterBlock.java
@@ -394,7 +394,14 @@ public class ComposterBlock extends Block implements WorldlyContainerHolder {
     @Override
     public void tick(BlockState state, ServerLevel world, BlockPos pos, RandomSource random) {
         if ((Integer) state.getValue(ComposterBlock.LEVEL) == 7) {
-            world.setBlock(pos, (BlockState) state.cycle(ComposterBlock.LEVEL), 3);
+            // Kiterino start
+            var newState = state.cycle(ComposterBlock.LEVEL);
+            var event = new org.bukkit.event.block.BlockFormEvent(org.bukkit.craftbukkit.block.CraftBlock.at(world, pos), org.bukkit.craftbukkit.block.CraftBlockStates.getBlockState(pos, newState, null));
+            if (!event.callEvent()) {
+                return;
+            }
+            world.setBlock(pos, newState, 3);
+            // Kiterino end
             world.playSound((Player) null, pos, SoundEvents.COMPOSTER_READY, SoundSource.BLOCKS, 1.0F, 1.0F);
         }
 
